schemas/2015-08-01/Microsoft.Web.json

```diff
@@ -0,0 +1,389 @@
+                                },
+                                {
+                                    "enum": [
+                                        "Free",
+                                        "Shared",
+                                        "Basic",
+                                        "Standard"
+                                    ]
+                                }
+                            ],
+                            "description": "Microsoft.Web/serverfarms: Server farm sku."
+                        },
+                        "workerSize": {
+                            "oneOf": [
+                                {
+                                    "$ref": "http://schema.management.azure.com/schemas/2015-05-01/deploymentTemplate.json#/definitions/expression"
+                                },
+                                {
+                                    "enum": [
+                                        "Small",
+                                        "Medium",
+                                        "Large"
+                                    ]
+                                },
+                                {
+                                    "type": "integer",
+                                    "minimum": 0,
+                                    "maximum": 2
+                                }
+                            ],
+                            "description": "Microsoft.Web/serverfarms: The instance size."
+                        },
+                        "numberOfWorkers": {
+                            "oneOf": [
+                                {
+                                    "$ref": "http://schema.management.azure.com/schemas/2015-05-01/deploymentTemplate.json#/definitions/expression"
+                                },
+                                {
+                                    "type": "integer",
+                                    "minimum": 0,
+                                    "maximum": 10
+                                }
+                            ],
+                            "description": "Microsoft.Web/serverfarms: The instance count, which is the number of virtual machines dedicated to the farm. Supported values are 1-10."
+                        }
+                    }
+                }
+            },
+            "required": [
+                "type",
+                "apiVersion",
+                "properties"
+            ]
+        },
+        "config": {
+            "type": "object",
+            "properties": {
+                "type": {
+                    "enum": [
+                        "Microsoft.Web/sites/config",
+                        "config"
+                    ]
+                },
+                "apiVersion": {
+                    "enum": [
+                        "2015-08-01"
+                    ]
+                },
+                "name": {
```

Comment:
With the name property enum here, the intellisense is better. The experience is that the user sees 3 choices for name, instead of just an empty string. Together with the name patterns on the web, connectionstrings and appsettings.schemas, it pivots the intellisense onto the right schema. If user selects web, they only see web specific properties, same goes for the other 2.